docs/guides/istio.md

```diff
@@ -1,39 +1,306 @@
 
-<<< @/examples/kubernetes/istio/virtual-services.yml
 
-#### Service Entry
+    The second resource is an `AuthorizationPolicy`, and it species:
+    - For objects with the `app.kubernetes.io/name` label matching `nginx`, only allow requests:
+      - **if** the request includes a JWT (already validated by `RequestAuthentication`) with the audience key `aud`,
+      - **and** the value of the `aud` key matches our known route, `hello.localhost.pomerium.io`.
 
-If you are enforcing mutual TLS in your service mesh you will need to add a ServiceEntry for your identity provider so that Istio knows not to expect a mutual TLS connection with, for example `https://yourcompany.okta.com`.
+    In other words, `RequestAuthentication` confirms that the incoming traffic to the sidecar has a signed and valid JWT, which confirms that the user has been authenticated and is authorized to access this service. The `AuthorizationPolicy` confirms that the traffic originated from a valid Pomerium route. The latter is especially important in Pomerium Enterprise, where a manager of a separate [Namespace](/enterprise/concepts.md#namespaces) could create a second route to a service normally routed and managed in your namespace.
 
-<<< @/examples/kubernetes/istio/service-entry.yml
+1. Apply the new resources with `kubectl`:
 
-#### Pomerium Configuration
+    ```bash
+    kubectl apply -f authorization-policy.yaml
+    ```
 
-For this example we're using the Pomerium Helm chart with the following `values.yaml` file. Things to note here are the `insecure` flag, where we are disabling TLS in Pomerium in favor of the Istio-provided TLS via sidecars. Also note the `extaEnv` arguments where we are asking Pomerium to extract the email property from the JWT and pass it on to Grafana in a header called `X-Pomerium-Claim-Email`. We need to do this because Grafana does not know how to read the Pomerium JWT but its auth-proxy authentication method can be configured to read user information from headers. The policy document contains a single route that will send all requests with a host header of `https://grafana.yourcompany.com` to the Grafana instance running in the monitoring namespace. We disable ingress because we are using the Istio ingressgateway for ingress traffic and don't need the Pomerium helm chart to create ingress objects for us.
+1. Now when you go to `hello.localhost.pomerium.io` in the browser, you should see `RBAC: access denied`. This confirms that the policy is in place and denying our request. To allow the traffic, add the `pass_identity_headers` annotation to `example-ingress.yaml`:
 
-<<< @/examples/kubernetes/istio/pomerium-helm-values.yml
+    ```yaml{7}
+    apiVersion: networking.k8s.io/v1
+    kind: Ingress
+    metadata:
+      name: hello
+      annotations:
+        cert-manager.io/issuer: pomerium-issuer
```

Comment:
@travisgroth didn't we set pomerium to be the default Ingress class already? If not, should we do that instead?

Reply:
I don't think we did (if I'm wrong can you link where?) and it would be better for demo purposes to only apply our ingress controller to selective resources rather than deploy as the default out of the box